hooks/useExchange.ts

```diff
@@ -106,18 +103,17 @@ const useExchange = ({
 	const quoteCurrencyKey = useRecoilValue(quoteCurrencyKeyState);
 
 	const [openModal, setOpenModal] = useState<ExchangeModal>();
-
-	const [isApproving, setIsApproving] = useState(false);
-	const [isApproved, setIsApproved] = useState(false);
+	const isApproved = useRecoilValue(isApprovedState);
+	const isApproving = useRecoilValue(isApprovingState);
 	const [gasInfo, setGasInfo] = useState<{ limit: number; l1Fee: Wei } | null>();
+	const [isSubmitting, setIsSubmitting] = useState(false);
+	const [atomicExchangeSlippage] = useState('0.01');
```

Comment:
Not from your changes but as this doesn't have a setter should we extract it to just be a constant `ATOMIC_EXCHANGE_SLIPPAGE`